regionmask/defined_regions/from_geopandas.py

```diff
@@ -0,0 +1,126 @@
+        )
+    abbrevs = []
+    for item in geodataframe.T:
+        name_for_abbrev = geodataframe.loc[item][names]
+        # catch region with no name
+        if name_for_abbrev is None:
+            name_for_abbrev = "UND"  # for undefined
+        abbrev = "".join(word[0] for word in name_for_abbrev.split(" "))
+        # remove certain chars
+        for char_to_remove in ["-", " ", ")", "("]:
+            abbrev = abbrev.replace(char_to_remove, "")
+        # if find duplicates, add counter
+        counter = 2
+        if abbrev in abbrevs:
+            while (abbrev + str(counter)) in abbrevs:
+                counter += 1
+            abbrev = abbrev + str(counter)
+        abbrevs.append(abbrev)
+    return abbrevs
+
+
+def from_geopandas(
+    geodataframe, numbers=None, names=None, abbrevs=None, name="unnamed", source=None
+):
+    """
+    Create ``regionmask.Region`` from ``geopandas.geodataframe.GeoDataFrame``.
+
+    Parameters
+    ----------
+    geodataframe : geopandas.geodataframe.GeoDataFrame
+        Shapefile to be downloaded. File has extensions ``.shp`` or ``.zip``.
+    numbers : str (optional)
+        Number of the column in shapefile that gives a region its number.
+        This column shouldnt have duplicates. If None (default), takes
+        ``geodataframe.index.values``.
+    names : str
+        Name of the column in shapefile that names a region. Breaks for duplicates.
+    abbrevs : str
+        Name of the column in shapefile that five a region its abbreviation.
+        Breaks for duplicates. If ``construct``, a combination of the first letters of
+        region name is taken.
+    name : str (optional)
+        name of the ``regionmask.Region`` instance created
+    source : str (optional)
+        source of the shapefile
+
+    Returns
+    -------
+    regionmask.core.regions.Regions
+
+    """
+    # get necessary data for Regions
+
+    if numbers is not None:
+        # sort, otherwise breaks
+        geodataframe = geodataframe.sort_values(numbers)
+        # ensure integer
+        geodataframe[numbers] = geodataframe[numbers].astype("int")
+        numbers = _maybe_get_column(geodataframe, numbers)
+    else:
+        numbers = geodataframe.index.values
+    # make sure numbers is a list
+    numbers = np.array(numbers)
+
+    if abbrevs is not None:
+        if abbrevs == "construct":
+            abbrevs = _construct_abbrevs(geodataframe, names)
+        else:
+            abbrevs = _maybe_get_column(geodataframe, abbrevs)
+    else:
+        raise ValueError(
+            "Please provide a string take can be taken from "
+            "geodataframe.columns for `abbrevs`, "
+            "found {}".format(abbrevs)
+        )
+    if names is not None:
+        names = _maybe_get_column(geodataframe, names)
+    else:
+        raise ValueError(
+            "Please provide a string take can be taken from "
+            "geodataframe.columns for `names`, found {}".format(names)
+        )
```

Comment:
Could you:
- remove the `else` branch
- add the two `checks`
- move this before the abbrevs (so we are sure the names are sane before constructing abbrevs)

Reply:
implemented